cpp/src/arrow/sparse_tensor.cc

```diff
@@ -134,6 +134,10 @@ inline Status CheckSparseCOOIndexValidity(const std::shared_ptr<DataType>& type,
 
   RETURN_NOT_OK(internal::CheckSparseIndexMaximumValue(type, shape));
 
+  // Indexes with no values are considered valid
```

Comment:
I don't understand what is happening here (I don't understand much about the tensors so not surprising).  Why is it ok if one of the shape elements is zero?  I would expect an empty sparse matrix to still have a shape:

```
>>> scipy.sparse.coo_matrix(numpy.zeros((2,4)), dtype=numpy.float32).shape
(2, 4)
```